R/check_outliers.R

```diff
@@ -48,29 +57,39 @@ check_outliers <- function(df,kobo_tool_location=NULL,
   #######################
 
 
-  df <- type.convert(df, as.is = TRUE,na.string= c(""," ")) |> rename(
+  df <- type.convert(df, as.is = TRUE,na.string= c(""," ")) |> dplyr::rename(
     uuid = !!rlang::sym(uuid_col_name)
   )
 
-  columns_to_add <- c(columns_to_add,"uuid") |> unique()
+  cols_to_add_cleaning_log <- c(cols_to_add_cleaning_log,"uuid") |> unique()
+
+
+
+  if(remove_choice_multiple == T ){
+    all_select_multiple_parent <- auto_sm_parent_children(df,sm_sep = sm_sep)
+    all_select_multiple_cols <-all_select_multiple_parent$sm_child
+  }
+
+  if(remove_choice_multiple == T ){columns_to_remove <- c(columns_to_remove,all_select_multiple_cols)}
+
+  cols_to_remove<- columns_to_remove[!columns_to_remove %in% cols_to_add_cleaning_log]
+
+
 
-  cols_to_remove<- columns_to_remove[!columns_to_remove %in% columns_to_add]
 
   if(!is.null(cols_to_remove)){
     df <- df %>% dplyr::select(-dplyr::all_of(cols_to_remove))
   }
 
 
 
-  if(!is.null(kobo_tool_location)) {
+  if(!is.null(kobo_survey) & !is.null(kobo_choices)) {
 
-    survey_sheet <- openxlsx::read.xlsx(kobo_tool_location,sheet = "survey")
-    choice_sheet <- openxlsx::read.xlsx(kobo_tool_location,sheet = "choices")
-    survey_sheet$name <- survey_sheet$name %>% stringr::str_replace_all("-",".")
+    kobo_survey$name <- kobo_survey$name %>% stringr::str_replace_all("-",".")
 
 
-    interger_column_in_kobo <- (survey_sheet %>% dplyr::filter(type == "integer") %>%
-                                  filter( !grepl('enumerator|_instance_', name)))$name
+    interger_column_in_kobo <- (kobo_survey %>% dplyr::filter(type == "integer") %>%
```

Comment:
Nothing to worry about but just a small misspelling of interger in lines 91 and 94.